test/Library.test.ts

```diff
@@ -321,5 +324,146 @@ describe('library as property or return or param', () => {
 
     });
   })
+
+  describe('Library as state property', () => {
+    describe('LibAsState1 test', () => {
+      let instance, result;
+      const Test = buildContractClass(loadDescription('LibAsState1_desc.json'));
+      const { L, ST } = buildTypeClasses(Test);
+      let l = new L(1, new ST({
+        x: 1,
+        c: true,
+        aa: [1,1,1]
+      }));
+      before(() => {
+        instance = new Test(l);
+      });
+
+      it('should success when call unlock', () => {
+
+        let newLockingScript = instance.getNewStateScript({
+          l: l.getNewState({
+            x: 6,
+            st: new ST({
+              x: 1,
+              c: false,
+              aa: [1,1,1]
+            })
+          })
+        })
+
+        const tx = newTx(inputSatoshis);
+        tx.addOutput(new bsv.Transaction.Output({
+            script: newLockingScript,
+            satoshis: outputAmount
+        }))
+  
+        const preimage = getPreimage(tx, instance.lockingScript, inputSatoshis)
+  
+        instance.txContext = {
+            tx: tx,
+            inputIndex,
+            inputSatoshis
+        }
+  
+        result = instance.unlock(1, new SigHashPreimage(toHex(preimage))).verify()
+        expect(result.success, result.error).to.be.true
+  
```

Comment:
Add a few more calls after it.

Reply:
tracked in this issue https://github.com/sCrypt-Inc/scryptlib/issues/163